### PR TITLE
{lib}[foss/2020b,fosscuda/2020b] TensorFlow v2.5.0 w/ Python 3.8.6

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusi8001 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/513c9fe34a949af01fd9f11fd66248de for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2432"
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3305.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/9457131249fbb51bd32ef58c1d269532 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2432
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3506.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/bce2f66b7885df48cce93f670c2accf5 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2432
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bask-pg0308u12a - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz (icelake), Python 3.6.8
See https://gist.github.com/a5761523d1cc15cd8fe1ebea9822480b for a full test report.

**Comment by Flamefire** [comment]
Thanks for the tests. It seems that 2 CPU tests are failing for @boegel and @branfosj  (those are the same 2). I'll add a foss version so I can test those too and see if it fails for me too. Strangely: It seems that the tests pass on AMD systems but fail on Intel systems :D

@boegel In your test report it says that 2 tests failed to build but the output of those is missing. Can you attach the full log?

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in total)
bask-pg0308u11a - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz (icelake), Python 3.6.8
See https://gist.github.com/ebeb693075c130725287bc02af4974cc for a full test report.

**Comment by boegel** [comment]
> @boegel In your test report it says that 2 tests failed to build but the output of those is missing. Can you attach the full log?

I don't have that log anymore, but I'll retry it and ensure the log doesn't go lost...

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (2 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/7aae6077f4b2be347608f3d05209424b for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
taurusi8008 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/4b54e711a2cb111dbdf6f729a0b9a4fb for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi8008 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/893dc07d5f6e5b761b74fbf2b4c8a273 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/ec934b83c0f8878a94bd6ddb1c38be1e for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusml23 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 2.7.5
See https://gist.github.com/29a801ee5a0bd14edca511bb369ef83a for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u12b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/4776edbdfea37b2802382dd516016045 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bber0501u03b.bb2.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/cc1523f51e8713fd1aa180a2de69063a for a full test report.

**Comment by Flamefire** [comment]
@branfosj Cascadelake again. This is starting to become a pattern... Anyway: IMO safe to ignore, more or less. Or don't build with native on cascadelake

**Comment by branfosj** [comment]
> @branfosj Cascadelake again. This is starting to become a pattern... Anyway: IMO safe to ignore, more or less. Or don't build with native on cascadelake

Yes, it looks like those two test failures happen on newer Intel CPUs. I'm seeing those failures on cascadelake and icelake. And the pass from my haswell node rules out the problem being caused by our setup (the cascadelake and haswell nodes should have identical OS packages, etc.).

**Comment by boegel** [comment]
Upstream issue for the failing tests on Cascade Lake: https://github.com/tensorflow/tensorflow/issues/49944

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
node3302.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/82b00ff9e574971456d10c60767aa64e for a full test report.

**Comment by boegel** [comment]
@Flamefire Do you think it makes sense to skip `quantized_ops_test`, or is that too big of a hammer?

**Comment by Flamefire** [comment]
@boegel IMO that failure is a valid failure, i.e. when optimizing for cascadelake there will be a real issue. So you either have to optimize for something older or explicitely skip the tests and accept the consequences. So until we have a patch I think having this fail is kinda important.
On the other hand we skipped the unique-string test on 2.4.1 which failed in the same way, but there we didn't know better.

Bottom line: This is (as far as I can tell) not a faulty test, which I'd skip, but a proof for an actual issue. So I would not ignore it.
But I can add a comment into the EC.

What do you say?

**Comment by boegel** [comment]
@Flamefire Yes, I agree. But then we should easily support installing TensorFlow with ignoring failing tests (which would produce a big fat warning, but include the test results in the log), rather than skipping all tests?

**Comment by Flamefire** [comment]
@boegel So add a `--ignore-failing-tests` to framework which catches `EasyBuildError` in the test_step and could be handled explicitely by easyblocks? Maybe in conjunction with a `report_test_failure` function in framework which either raises an error or `print_warning`s it?

**Comment by boegel** [comment]
> @boegel So add a `--ignore-failing-tests` to framework which catches `EasyBuildError` in the test_step and could be handled explicitely by easyblocks? Maybe in conjunction with a `report_test_failure` function in framework which either raises an error or `print_warning`s it?

Something like that, yes. And I would make it clash with `--skip-test-step`, using both doesn't make sense and should be disallowed.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi8022 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/0308429315c6127b0630d8f10eac46f4 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/08ecdf25c08051ecfa35715c14605fdc for a full test report.

**Comment by boegel** [comment]
Quick update after rebuilding `TensorFlow-2.5.0-foss-2020b.eb` a bunch of times today:

* the failing `quantize` tests are also happening on Intel Skylake, not just on Cascade Lake...
* disabling `optarch` fixes the issue:
   ```python
   toolchainopts = {'pic': True, 'optarch': False}
   ```
* building with `--optarch=march=haswell` (and leaving the `optarch` toolchain option enabled) also fixes the broken tests;
* selectively disabling a specific subset of the AVX512 instructions also fixes the tests:
   ```
   toolchainopts = {'pic': True, 'extra_cxxflags': '-mno-avx512f'}
   ```
    I narrowed this down after starting with `-march=native -mno-avx512f -mno-avx512pf -mno-avx512er -mno-avx512cd -mno-avx512vl -mno-avx512bw -mno-avx512dq` which is essentially equivalent with `-march=haswell`.

This testing was only done on Intel Skylake for now, I'll revisit this on Cascade Lake and also try the `fosscuda` easyconfig.

If that works out well, I think we can selectively inject this flag on all `x86_64` systems in the easyconfig, as follows:

```python
if ARCH == 'x86_64':
    toolchainopts['extra_cxxflags'] = '-mno-avx512f'
```

Or we can add something to the TensorFlow easyblock for this, perhaps, where we can do it a bit more dynamically based on CPU features for example, rather than for all `x86_64` CPUs.

@branfosj Any chance you can try with `-mno-acx512f` on your Cascade Lake (or even Ice Lake) system to verify this workaround?

**Comment by branfosj** [comment]
Yes, adding `'extra_cxxflags': '-mno-avx512f'` to `toolchainopts` on my CasadeLake and it passes the tests (`TensorFlow-2.5.0-foss-2020b.eb`).

I've also set off tests on IceLake for both the `foss` and `fosscuda` versions. I'll report on those later.

**Comment by boegel** [comment]
@Flamefire Can we add this to the easyconfigs for TensorFlow 2.5.0, right below the `toolchainopts =` line?

```python
if ARCH == 'x86_64':
    # required workaround for failing tests on Intel Skylake and Cascade Lake,
    # see https://github.com/tensorflow/tensorflow/issues/49944
    toolchainopts['extra_cxxflags'] = '-mno-avx512f'

**Comment by Flamefire** [comment]
Done.

**Comment by branfosj** [comment]
The tests pass (both foss and fosscuda versions of the EC) on IceLake with the flag set. I've now set off test reports for this PR.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u17b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/602987bb20f5a2c3c9d138ca962bd8d8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/d23d52c7f502c9e10fa35c7d79cab545 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2617.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/5c2d44c52d30a8dac0feff745cf0fb14 for a full test report.

**Comment by Flamefire** [comment]
@boegel They are really annoying. Failure on ARM is due to https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/lite/kernels/internal/optimized/depthwiseconv_3x3_filter_common.h#L125-L127

The function is `vdotq_lane_s32 (int32x4_t __r, int8x16_t __a, int8x8_t __b, const int __index)`, so the comment is plain wrong.

Can you open an issue in the TF repo?

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bask-pg0309u15a - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz (icelake), Python 3.6.8
See https://gist.github.com/1c529a5fa116c6546ca12a5026744eda for a full test report.

**Comment by Flamefire** [comment]
@boegel I added a patch for ARM (and reordered the source-patch-checksum stuff), so can you restart on ARM? No need to rerun any others as the patch does not affect other archs

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3110.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/670fe13b9417d1dada4e5e3ad3f3eeb4 for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.5.0_fix-arm-vector-intrinsics.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
+The comment is not true, the function actually takes the arguments as it should
+Hence just redefine the function
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.5.0_fix-arm-vector-intrinsics.patch:2]: @Flamefire Are you planning to open an upstream fix/PR for this?

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bask-pg0309u16a - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz (icelake), Python 3.6.8
See https://gist.github.com/65782114003be830c2efc44b2e276d29 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3550.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/d64f55c2fb8de0c8ec32a7d086a595da for a full test report.

**Comment by boegel** [comment]
> @boegel I added a patch for ARM (and reordered the source-patch-checksum stuff), so can you restart on ARM? No need to rerun any others as the patch does not affect other archs

Thanks, new test report coming up!

To be clear: I don't consider failing tests on `aarch64` a blocker for this PR (see also #12667, for example).

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="TensorFlow-2.5.0-foss-2020b.eb"
CORE_CNT=16

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.5.0_fix-arm-vector-intrinsics.patch (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.5.0_fix-arm-vector-intrinsics.patch:2]: Not really. I guess that comment is true for their environment. Could you open an upstream issue with that? They need details about the environment and as we don't have an ARM cluster I can't really contributed there.

Might also be worth testing this fix first and mentioning it if it does work. I was just guessing here.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/6e931318a7e5eac4f08296ede25a2dac for a full test report.

**Comment by Flamefire** [comment]
@boegel please retry with https://github.com/easybuilders/easybuild-easyconfigs/pull/13086
You are hitting https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93235

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3308.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/1fdb3af2ed5936e1d0b198442bf089b2 for a full test report.

**Comment by boegel** [comment]
> @boegel please retry with ~#13086~
> You are hitting https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93235

New test is running on `aarch64` after rebuilding `GCCcore/10.2.0` with the fix from ~#13086~...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/a99433ec91f14371d833916fb58ea205 for a full test report.

**Comment by Flamefire** [comment]
@boegel OOM?
> Killed signal terminated program cc1plus

**Comment by boegel** [comment]
> @boegel OOM?
> 
> > Killed signal terminated program cc1plus

Yes, using only half the cores lets the build pass. Test step is still running...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
select-pika-c6gd-2xlarge-0001 - Linux centos linux 8.3.2011, AArch64, ARM UNKNOWN (graviton2), Python 3.6.8
See https://gist.github.com/79896f69ec4e7b7eb2a0d38c2e1a3410 for a full test report.

**Comment by Flamefire** [comment]
@boegel The TF support for ARM is broken. Please open a bug with them. The most simple one is `SparseMatmulOpTest.BroadcastPacketTest`
> [0.170094 0.170094 0.170094 0.170094] != [  0.170094    0.14922 -0.0823886   0.026985]

This is because they use https://github.com/tensorflow/tensorflow/blob/5dcfc51118817f27fad5246812d83e5dccdc5f72/tensorflow/core/kernels/sparse_matmul_op.h#L93-L95 which basically simply returns the vector of 4 values instead of broadcasting the first value which is incredibly stupid -.-

The other ones are likely related, see `TensorSliceWriteTest.CheckpointSize`:
>   TensorSliceWriter::MaxBytesPerElement(DT_HALF)
            Which is: 3
          BytesPerElementHelper&lt;Eigen::half&gt;(Eigen::half(-1.0))
            Which is: 10

This hints at either a bug in Eigen or their treatment of the Eigen::half type. So fixing that should fix the lot.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild2.novalocal - Linux centos linux 8.3.2011, POWER, IBM pSeries (emulated by qemu) (power9le), Python 3.6.8
See https://gist.github.com/2f30f6e108911ef66675b1e139b92900 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3303.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/97e8257076f99e0a1fe2a931ccc8debe for a full test report.

**Comment by Flamefire** [comment]
@boegel Not sure what went wrong on generoso, would need the log as the failing test isn't in the report. I suspect a compile error.

My suggestion would be to merge this as it passes on a few platforms and follow-up with fixes as required.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3600.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/b51e8e7e4e1cf6d735c1d96162278b48 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3302.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/cb8bb48db97a4ffcc01175119f50a968 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3168.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/8841f0394cbcc21f7eac1394d0c5f3bd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2618.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/89c727e6d8e702d11447579c96dea160 for a full test report.

**Comment by Flamefire** [comment]
Added the patch from https://github.com/easybuilders/easybuild-easyconfigs/pull/13181 just to be safe. Impact for existing stuff is very low, it "just" avoids potential for misoptimization. Reports coming up just to make sure format etc. matches and it builds at all. But I don't think you need to redo your tests for that patch.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi8029 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/4331a1464784330b5ed53d02be62072f for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3302.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/a7f8a3cc2578c539eaa5af98c39bd9e5 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusa7 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/76d70f8016a1cf1d11af5d6ffc0fc71f for a full test report.

**Comment by Flamefire** [comment]
> OSError: [Errno 28] No space left on device

On generoso ;)

PS: I got a PR against TF easyblock which improves error reporting and fixes the python package detection issue.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!
